### PR TITLE
Fix parameter ordering so that it's consistent across sessions.

### PR DESCRIPTION
**Pull request by Bonnevie**
Fixes issue #416 by changing the sorting key (currently based on `id`) to something that is static between sessions (`Param.name` in this PR, picked slightly arbitrarily).

### CODE REVIEW

File: GPflow/param.py [hunk 1]
```diff
@@ -796,7 +796,7 @@ def sorted_params(self):
         params=  [child for key, child in self.__dict__.items()
                   if isinstance(child, (Param, Parameterized)) and
                   key is not '_parent']
-        return sorted(params, key=id)
+        return sorted(params, key=lambda x: x.name)
```
Comment (john-bradshaw) [inline GPflow/param.py]: I think you want to use `long_name`?

Comment (Bonnevie) [inline GPflow/param.py]: yup, was just fixing it.

**Comment by Bonnevie** [comment]
From the issue thread:

Running a timing experiment in IPython (using %magic) of the form

```
class A():
    def __init__(self, N = 20):
        self.long_name = ''.join(random.choice(string.ascii_uppercase +
                                 string.digits) for _ in range(N))

L = [A() for _ in range(100)]
test_id = lambda: sorted(L, key=id)
test_name = lambda: sorted(L, key=lambda x: x.long_name)
%timeit test_id
%timeit test_name
```
results in timings with almost no difference, and if anything, the timings are in favor of the long_name key.

edit: e.g. one run gives 
```
10000000 loops, best of 3: 29.1 ns per loop
10000000 loops, best of 3: 24.8 ns per loop

```

**Comment by jameshensman** [comment]
You need to add parentheses after the functions in %timeit. 

I get 
```python
In [14]: %timeit test_name()
10000 loops, best of 3: 27.7 µs per loop

In [15]: %timeit test_id()
100000 loops, best of 3: 10.3 µs per loop
```

**Comment by Bonnevie** [comment]
@jameshensman: ah, sorry about that. I can see how looking up the function name alone would be a pretty fast operation :)
Still doesn't seem like a huge time difference?

**Comment by javdrher** [comment]
This would be a helpful addition for GPflowOpt. Yesterday I was trying to implement BO with MCMC for the hypers by copying, using set_state(), and then adding all scores together. However in a (deep)copy of a model the order also differs.

**Comment by Bonnevie** [comment]
@jameshensman: there seems to be a few minor PRs stuck in limbo - is there anything further I should fix before this can be merged?

**Comment by jameshensman** [comment]
This PR seems good, but I think it requires a version bump. 

See https://github.com/GPflow/GPflow/blob/master/contributing.md

Could you handle that, @Bonnevie and then I will merge this.

**Comment by Bonnevie** [comment]
@jameshensman: was this what you had in mind?

**Pull request merged**